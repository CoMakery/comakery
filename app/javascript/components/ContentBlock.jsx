import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Block = styled.div`
  margin-bottom: 25px;
`

const Title = styled.div`
  font-family: Montserrat;
  font-size: 12px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #3a3a3a;
  text-transform: uppercase;
  margin-bottom: 5px;
`

const Children = styled.div`
  font-family: Georgia;
  font-size: 14px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #4a4a4a;
`

class ContentBlock extends React.Component {
  render() {
    return (
      <React.Fragment>
        <Block>
          <Title>
            {this.props.title}
          </Title>
          <Children>
            {this.props.children}
          </Children>
        </Block>
      </React.Fragment>
    )
  }
}

ContentBlock.propTypes = {
  title: PropTypes.string.isRequired
}
ContentBlock.defaultProps = {
  title: ''
}
export default ContentBlock
