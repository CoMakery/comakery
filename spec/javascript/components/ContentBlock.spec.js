import React from 'react'
import { mount } from 'enzyme'
import ContentBlock from 'components/ContentBlock'

describe('ContentBlock', () => {
  it('renders component correctly', () => {
    const wrapper = mount(<ContentBlock />)

    expect(wrapper).toMatchSnapshot()
  })
})
