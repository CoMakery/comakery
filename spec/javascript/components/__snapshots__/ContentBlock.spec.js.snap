// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentBlock renders component correctly 1`] = `
<ContentBlock
  title=""
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "hqxcBJ",
            "rules": Array [
              "
  margin-bottom: 25px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa hqxcBJ"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "kFENjx",
                  "rules": Array [
                    "
  font-family: Montserrat;
  font-size: 12px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #3a3a3a;
  text-transform: uppercase;
  margin-bottom: 5px;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-bwzfXH kFENjx"
            />
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": true,
                  "lastClassName": "bNkIaI",
                  "rules": Array [
                    "
  font-family: Georgia;
  font-size: 14px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #4a4a4a;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-htpNat bNkIaI"
            />
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</ContentBlock>
`;
